data_format_config.go

```diff
@@ -141,3 +143,44 @@ func (o *JSONOptions) Map() map[string]interface{} {
 	}
 	return map[string]interface{}{}
 }
+
+// XLSXOptions specifies configuraiton details for the xlsx file format
+type XLSXOptions struct {
+	SheetName string `json:"sheetName,omitempty"`
+}
+
+// NewXLSXOptions creates a XLSXOptions pointer from a map
+func NewXLSXOptions(opts map[string]interface{}) (FormatConfig, error) {
+	o := &XLSXOptions{}
+	if opts == nil {
+		return o, nil
+	}
+
+	if opts["sheetName"] != nil {
+		if sheetName, ok := opts["sheetName"].(string); ok {
+			o.SheetName = sheetName
+		} else {
+			return nil, fmt.Errorf("invalid sheetName value: %v", opts["sheetName"])
+		}
+	}
+
+	return o, nil
+}
+
+// Format announces the XLSX data format for the FormatConfig interface
+func (*XLSXOptions) Format() DataFormat {
+	return XLSXDataFormat
+}
+
+// Map structures XLSXOptions as a map of string keys to values
+func (o *XLSXOptions) Map() map[string]interface{} {
+	if o == nil {
```

Comment:
Do we need to check nil here? In the rest of codebase, we're assuming that interface types won't contains a nil pointer value, for example see lib/datasets.go. Seems like an anti-pattern to support it here, and the code will just crash a few lines below where we try to resolve o.SheetName.